sdk/framework/openapi.go

```diff
@@ -464,6 +465,21 @@ func specialPathMatch(path string, specialPaths []string) bool {
 func expandPattern(pattern string) []string {
 	var paths []string
 
+	// recursively operate on each pattern after determining a set of unique roots is present,
+	// e.g. "(rootOne|rootTwo)/(?P<name>regex)" should yield "rootOne/{name}", "rootTwo/{name}"
+	rootMatch := altRootsRe.FindAllString(pattern, -1)
+	if len(rootMatch) > 0 {
+		m := strings.Replace(rootMatch[0][1:], ")/", "", -1)
+		if m != "" {
+			var expandedRoots []string
+			suffix := strings.Replace(pattern, rootMatch[0], "", -1)
```

Comment:
There are a number of of [replace utilities](https://pkg.go.dev/regexp#Regexp.ReplaceAllString) for in regexp package, not sure if they would be useful here.

Reply:
Since `strings` was already used in the package I thought it was safe here. Are you worried about performance?

Reply:
I was more wondering if the code could be simplified a bit into something easy to verify at a glance. One thing that could help add clarity here is capturing each root expression (`rootOne`, `rootTwo`, etc.) into its own subgroup and then iterating over the matches to form expanded roots. This would remove the need to split the string manually as well as the `string.Replace` calls.